### PR TITLE
Enable proof checking for QF_LRA benchmarks

### PR DESCRIPTION
**Pull request by 4tXJ7f**
Due to issues in the current proof code, this commit also disables proof
checking for five QF_LRA benchmarks (see issue #2855). This commit also
fixes issues related to comparing `InfoLogic`s with/without higher-order
or cardinality constraints.

### CODE REVIEW

File: src/smt/smt_engine.cpp [hunk 1]
```diff
@@ -4431,7 +4431,7 @@ void SmtEngine::checkProof()
   if (!(
           // Pure logics
           logicString == "QF_UF" || logicString == "QF_AX"
-          || logicString == "QF_BV" ||
+          || logicString == "QF_BV" || logicString == "QF_LRA" ||
           // Non-pure logics
```
Comment (yoni206) [inline src/smt/smt_engine.cpp]: Any reason to not include `QF_UFLRA` and other combinations?

Comment (4tXJ7f) [reply src/smt/smt_engine.cpp]: Good point, I've enabled it now and fixed some issues with comparing `LogicInfo`s in the process.

**Comment by 4tXJ7f** [comment]
@ajreynol I've added you to the PR if you are interested to look at the changes in `LogicInfo` related to comparing logics with/without cardinalities/higher-order constraints.

**Comment by 4tXJ7f** [comment]
I've updated #2855 to include some additional failing benchmarks.

**Pull request merged**